src/main/scala/io/dronekit/CouchbaseStreamsWrapper.scala

```diff
@@ -259,43 +367,89 @@ class CouchbaseStreamsWrapper(host: String, bucketName: String, password: String
 
   /**
    * Query an index for multiple items
-   * @param designDoc The design document name to query
+    *
+    * @param designDoc The design document name to query
    * @param viewDoc The view in that design doc
    * @param keys The list of keys to query for
    * @param stale Allow potentially stale indexes or not
    * @return Observable of AsyncViewRow
    */
-  def indexQuery(designDoc: String, viewDoc: String, keys: List[String] = List(), stale: Stale = Stale.FALSE):
-  Observable[AsyncViewRow] = {
+  def indexQuery(designDoc: String, viewDoc: String, keys: List[String] = List(), stale: Stale = Stale.FALSE,
+                limit: Int = Int.MaxValue, skip: Int = 0): Observable[AsyncViewRow] = {
     // Couchbase needs a java.util.List
     val keyList: java.util.List[String] = keys
     val query =
-      if (keys.isEmpty) ViewQuery.from(designDoc, viewDoc).stale(stale).inclusiveEnd(true)
-      else ViewQuery.from(designDoc, viewDoc).stale(stale).inclusiveEnd(true).keys(JsonArray.from(keyList))
+      if (keys.isEmpty)
+        ViewQuery.from(designDoc, viewDoc)
+          .stale(stale)
+          .inclusiveEnd(true)
+          .limit(limit)
+          .skip(skip)
+      else
+        ViewQuery.from(designDoc, viewDoc)
+          .stale(stale)
+          .inclusiveEnd(true)
+          .keys(JsonArray.from(keyList))
+          .limit(limit)
+          .skip(skip)
     toScalaObservable(bucket.async().query(query))
       .flatMap(queryResult => queryResult.rows())
   }
 
   /**
    * Query an index with a compound key
-   * @param designDoc The name of the design document
+    *
+    * @param designDoc The name of the design document
    * @param viewDoc The name of the view
    * @param keys A List of lists to query for
    * @param stale Allow stale records
    * @return Observable of AsyncViewRows
    */
-  def compoundIndexQuery(designDoc: String, viewDoc: String, keys: List[List[Any]], stale: Stale = Stale.FALSE):
-  Observable[AsyncViewRow] = {
+  def compoundIndexQuery(designDoc: String, viewDoc: String, keys: List[List[Any]], stale: Stale = Stale.FALSE,
```

Comment:
Up to you I guess, you could write a new method, or modify the existing ones if it will be compatible with previous usage.